### PR TITLE
fix: hdr.Xattrs deprecated

### PR DESCRIPTION
**Pull request by testwill**

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Did you test this PR?

File: pkg/archive/archive.go [hunk 1]
```diff
@@ -500,8 +500,8 @@ func ReadSecurityXattrToTarHeader(path string, hdr *tar.Header) error {
 			capability[versionOffset] = vfsCapRevision2
 			length = xattrCapsSz2
 		}
-		hdr.Xattrs = make(map[string]string)
-		hdr.Xattrs["security.capability"] = string(capability[:length])
+		hdr.PAXRecords = make(map[string]string)
+		hdr.PAXRecords["security.capability"] = string(capability[:length])
```
Comment (AkihiroSuda) [inline pkg/archive/archive.go:504]: missing `SCHILY.xattr`

File: pkg/archive/archive.go [hunk 2]
```diff
@@ -776,7 +776,7 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, o
 	}
 
 	var xattrErrs []string
-	for key, value := range hdr.Xattrs {
+	for key, value := range hdr.PAXRecords {
```
Comment (corhere) [inline pkg/archive/archive.go:779]: Not all PAXRecords are xattrs. Applying all records as xattrs, especially without first stripping off the "SCHILY.xattr." prefix from the xattr name, is incorrect.

**Review by corhere — CHANGES_REQUESTED** [review]
@AkihiroSuda is right; [xattr PAXRecords are prefixed with "SCHILY.xattr."](https://pkg.go.dev/archive/tar#Header:~:text=//%20Xattrs%20stores%20extended%20attributes%20as%20PAX%20records%20under%20the%0A%09//%20%22SCHILY.xattr.%22%20namespace.%0A%09//%0A%09//%20The%20following%20are%20semantically%20equivalent%3A%0A%09//%20%20h.Xattrs%5Bkey%5D%20%3D%20value%0A%09//%20%20h.PAXRecords%5B%22SCHILY.xattr.%22%2Bkey%5D%20%3D%20value)

**Comment by neersighted** [comment]
Relevant, though it does not make the situation better or worse: https://github.com/opencontainers/image-spec/issues/725

**Pull request closed without merging**